regression/pages/common/utils.py

```diff
@@ -0,0 +1,496 @@
+        lambda: page.q(
+            css='select[name={}] option[value="{}"]'.
+            format(drop_down_name, value)
+        ).selected,
+        "Correct value is selected",
+        timeout=DEFAULT_TIMEOUT
+    )
+
+
+def click_checkbox(page, css_selector):
+    """
+    A helper function to click check boxes
+    Args:
+       page:
+       css_selector:
+    """
+    # pylint: disable=unnecessary-lambda
+    page.wait_for_element_visibility(css_selector, 'wait for target checkbox')
+    page_elem = page.q(css=css_selector).results[0]
+    if not page_elem.is_selected():
+        page.browser.execute_script("arguments[0].click();", page_elem)
+        page.wait_for(
+            lambda: page_elem.is_selected(),
+            'Target checkbox is selected'
+        )
+
+
+def remove_spaces_from_list_elements(list_with_spaces):
+    """
+    A helper function to remove spaces from all items in the list
+    Args:
+      list_with_spaces:
+    """
+    return [x.replace(' ', '') for x in list_with_spaces]
+
+
+def get_text_against_page_elements(page, css_selectors):
+    """
+    A helper function to get text values for a list of web elements
+    Args:
+        page:
+        css_selectors:
+    Returns:
+        list_of_text:
+    """
+    list_of_text = []
+    for css_selector in css_selectors:
+        list_of_text.append(page.q(css=css_selector).text[0])
+    return list_of_text
+
+
+def extract_mmm_dd_yyyy_date_string_from_text(original_string):
+    """
+    Extract date of MM dd, yyyy from text
+    Args:
+        original_string:
+    Returns:
+        date_string
+    """
+    regex_result = re.search(
+        r'(?:Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)\s\d{2},\s\d{4}',
+        original_string
+    )
+    if regex_result:
+        date_string = regex_result.group(0)
+        return date_string
+    else:
+        return 'Required date pattern not found in search string'
+
+
+def convert_date_format(original_date, original_format, required_format):
+    """
+    Convert dates format to required pattern
+    Args:
+       original_date:
+       original_format:
+       required_format:
+    Returns:
+       modified date:
+    """
+    try:
+        return datetime.datetime.strptime(
+            original_date,
+            original_format
+        ).strftime(required_format)
+    except ValueError:
+        return 'Invalid date or format'
+
+
+def extract_numerical_value_from_price_string(raw_price_string):
+    """
+    Extract numerical value from a string containing price
+    Args:
+        raw_price_string:
+    Returns:
+        numerical price:
+    """
+    regex_result = re.search(
+        r'\d+,\d+\.\d+|\d+\.\d+|\d+,\d+|\d+',
+        raw_price_string
+    )
+    if regex_result is not None:
+        price_value = regex_result.group(0)
+        return float(price_value.replace(",", ""))
+    else:
+        return 'No numerical value found in search string'
+
+
+def substring_from(original_string, delim):
```

Comment:
I'm wondering why you wouldn't just say something like `"hi" in "hello hi there"`. I'm not sure why you would want to build a helper function here.